### PR TITLE
fixed crash when encountering anonymous struct attributes

### PR DESCRIPTION
**Pull request by kofoworola**
Made gunk reject anonymoius/unamed attributes and added a human friendly error message when encountered. Also ran gofmt

### CODE REVIEW

File: loader/loader.go [hunk 1]
```diff
@@ -405,6 +405,14 @@ func (l *Loader) validatePackage(pkg *GunkPackage) {
 				return true
 			}
 
+			//look through all fields for anonymous/unnamed types
```
Comment (mvdan) [inline loader/loader.go]: space this comment, capitalise it, and add a period.

File: loader/loader.go [hunk 2]
```diff
@@ -405,6 +405,14 @@ func (l *Loader) validatePackage(pkg *GunkPackage) {
...
+			for _, field := range st.Fields.List {
+				if field.Names == nil || len(field.Names) < 1 {
```
Comment (mvdan) [inline loader/loader.go]: remove the nil check.

File: loader/loader.go [hunk 3]
```diff
@@ -405,6 +405,14 @@ func (l *Loader) validatePackage(pkg *GunkPackage) {
...
+					pkg.addError(ParseError, st.Pos(), l.Fset, "anonymous struct attributes are not supported")
```
Comment (mvdan) [inline loader/loader.go]: s/attributes/fields/

**Comment by mvdan** [comment]
Right, I completely forgot to spot that this is missing a test. You've written test scripts before, right? Please write a simple one for this crash. There might already be a testscript that covers a bunch of previous crashes, in which case just add to it.

**Comment by kofoworola** [comment]
@mvdan @mgfcmartinez added the test script. Kindly re-review

**Pull request merged**